### DISCUSSION TITLE
PHP8.3 Causing CPU to hit 100% for Apache Web Server on Windows 11

### DISCUSSION THREAD

**Discussion by teamayu** [leokhoa/laragon#663] [General]
Hello. I am a long time user of Laragon, thank you for this awesome software.

My problem is I have issues with PHP8.3. I have tried 8.3.1 and 8.3.2 and both of them keeping spiking the CPU to 100% on Windows 11, 64bit for the Apache Web Server. I've never had issue with Laragon and updating PHP, but something is wrong with the 8.3 branch. I have the latest Apache version , but same problem. It occasionally runs ok for a few mins, then the CPU hits 100% and it just hangs. 

The current Apache version I have is "Apache 2.4.58 Win64", the previous version I used was "Apache 2.4.41 Win64". 2.4.41 was throwing a runtime error, so upgraded as per the discussion here:

https://github.com/leokhoa/laragon/discussions/576

For now I'm sticking with with the 8.2 branch, which is fine. Anyone have any ideas what the problem could be? I'm using Laragon 6.

Thank you.

**Comment by RobiNN1** [comment]
I have the same problem. Even with all php extensions disabled. There is a spam of this error in the Apache error log `VirtualProtect() failed [87] The parameter is incorrect`.

**Reply by xarris88** [reply]
Hello, 

check if the OPCache is enabled in php.ini. If yes, try if disabling it and check if this fixes the issue.

**Reply by RobiNN1** [reply]
As i wrote "Even with all php extensions disabled.". Ofc. I checked this.

**Reply by xarris88** [reply]
oops, even reading the message twice I missed it. Sorry. 
(This was my issue)

**Comment by teamayu** [comment]
I also disabled the extensions, so OPCache isn't an issue. I was wondering if it was related to PHP8.3.1, so I waited for PHP8.3.2 which is the same, so not sure where the issue lies at the moment. Bug with Apache 2.4.58? Just wondered as previous versions didn't seem to have issues.

**Reply by KieranFJ** [reply]
Have you got Xdebug enabled?

There seems to be an issue with xdebug being enabled, PHP8.3.

**Reply by teamayu** [reply]
Correct, XDebug is indeed the issue. Just upraded to 8.3.3 ts and it's fine until you enable XDebug.

Thanks for the help.

**Reply by teamayu** [reply]
It has been reported on the XDebug site:

https://bugs.xdebug.org/view.php?id=2233

https://bugs.xdebug.org/view.php?id=2216

**Comment by xarris88** [comment]
Just to mention my setup in case that you haven't tried the very same configuration:
 Laragon 6.0 220916
 Apache 2.4.58
 PHP 8.3.2 Non Thread Safe VS16 x64
 mod_fcgid-2.3.10-Win64-vs17.so (etc/apache2/modules)

**Reply by RobiNN1** [reply]
So there is issue.. `Thread Safe (Apache 2.0 Handler)` vs `Non Thread Safe (CGI/FastCGI)`.

I've never used `nts` version so I completely forgot to try it. Anyway thanks, as temporary solution it's fine.

**Comment by teamayu** [comment]
> So there is issue.. `Thread Safe (Apache 2.0 Handler)` vs `Non Thread Safe (CGI/FastCGI)`.
> 
> I've never used `nts` version so I completely forgot to try it. Anyway thanks, as temporary solution it's fine.

Same here, thanks xarris88. Seems to have worked great. And I agree, as a temp solution this will be fine.

**Comment by luisAntonioLAGS** [comment]
It hasn't given me any problems in Windows 10 Pro for workstations.

I tried it on Windows 11 and it hasn't given me any problems either.

Screenshot windows 10 (It is the OS I use at my current job)
![imagen](https://github.com/leokhoa/laragon/assets/60680854/59d9f573-a1ce-4312-abf5-23da53fd644d)

Screenshot windows 11
![imagen](https://github.com/leokhoa/laragon/assets/60680854/1f7fa12c-9c8a-46d4-aec8-7df458da182f)

**Comment by NitemareReal** [comment]
Same problem for me, under Windows 10 and Windows 11. Disabling XDebug, Apache works fine.

**Comment by AmirKhan47** [comment]
![image](https://github.com/leokhoa/laragon/assets/23741716/890ab299-cbce-4973-88c9-b27655ab9203)
I am facing an issue with this on the latest version.

**Comment by RobiNN1** [comment]
https://xdebug.org/download/historical#3_3_2

~~Its fixed in dev version https://github.com/xdebug/xdebug/pull/958~~

~~You can download dll from Artifacts https://github.com/xdebug/xdebug/actions/runs/8475033532~~

**Reply by michelinok** [reply]
Can you please help me choosing the right dll?

**Reply by RobiNN1** [reply]
There is already a stable version 3.3.2 with a fix for this https://xdebug.org/download/historical#3_3_2

**Comment by tripledm** [comment]
On a new installation with only php 8.3 I also get high cpu usage it maxes out the server frequently. A week later and few tries of solutions I realised on GCP install http/2 not enabled but that did not fix it either. I rolled back some of the sites on there to php 8.2 and presto, huge drop in CPU usage already and none of the website are busy or in high load periods. Funny thing is only 2 of 15 dropped and it resolved it so not sure how to work that one out but temp solution I have at least.
GCP single VM instance with AlmaLinux v9.4.0 STANDARD kvm, cPanel Version [122.0.21] for the record.
